pkg/metrics/types.go

```diff
@@ -86,6 +88,9 @@ var (
 	// DNS Metrics.
 	DNSRequestCounter  ICounterVec
 	DNSResponseCounter ICounterVec
+
+	InfinibandCounterStats IGaugeVec
```

Comment:
this is named CounterStats but is an IGaugeVec type, would an ICounterVec be better served for this?

Reply:
I considered this but thought on plugin startup the IB counter will not be reset at that time so the metric would need to be a gauge to assume the current value. Let me know if  I should be approaching this differently.

Reply:
Gauge and Counter have different semantic meanings and the data streams they produce can be processed in different ways. Additional context on this in #237.

The underlying data here is monotonically increasing - it never goes down - and counts up events forever.

Reply:
Yes, I understand the difference between the metric types but I don't think it addresses the issue I raised. Please let me know if it's not clear. My thought is that if the plugin counter isn't in sync with the data source then the plugin counter won't increment in a meaningful way.